### PR TITLE
Enable evaluation metrics calculation functions to consume tables 

### LINKED ISSUE #50: Lighthouse should consume tables
This is the counterpart to #45 which is about producing rows. It would be nice if the inputs were tables.

- [ ] Create Legolas schema with 1 column per segment with columns for what we need in order to evaluate (just votes, predicted_soft_labels, elected_hard_labels?)
- [ ] Write an `evaluation_metrics` method that consumes tables with this schema, along with keyword arguments (classes, thresholds, probably?)

### PR DESCRIPTION
**Pull request by hannahilea**
Closes #50.

To do:
- [x] Basic schema implementation
- [x] roundtrip tests
- [x] clean up usage of `nothing` v `missing` in threshold
- [x] Breaking version bump? (Maybe/maybe not---depends if API changes around `nothing` -> `missing`)---add support for both options, don't remove original support for one or the other
- [x] Field renaming in `ObservationRow`---change at all?

### CODE REVIEW

File: src/row.jl [hunk 1]
```diff
@@ -131,3 +136,54 @@ Lighthouse <v0.14.0).
...
+"""
+const ObservationRow = Legolas.@row("lighthouse.observation@1",
+                                    predicted_hard_label::Int64,
+                                    predicted_soft_labels::Vector{Float32},
+                                    elected_hard_label::Int64,
+                                    votes::Union{Missing,Vector{Int64}})
+
+function _observation_table_to_inputs(observation_table)
+    Legolas.validate(observation_table, OBSERVATION_ROW_SCHEMA)
+    df_table = DataFrame(observation_table; copycols=false)
```
Comment (ericphanson) [inline src/row.jl:164]: I think we don't actually need a `DataFrame`, just a column table:
```suggestion
    df_table = Tables.columns(observation_table)
```
(DataFrame additionally supports `transform`, `select`, etc, but here we just want to access columns)

File: src/row.jl [hunk 2]
```diff
@@ -131,3 +136,54 @@ Lighthouse <v0.14.0).
...
+function _inputs_to_observation_table(; predicted_hard_labels::AbstractVector,
+                                      predicted_soft_labels::AbstractMatrix,
+                                      elected_hard_labels::AbstractVector,
+                                      votes::Union{Nothing,Missing,AbstractMatrix}=nothing)
+    votes = has_value(votes) ? collect(eachrow(votes)) : missing
+    observations = DataFrame(; predicted_hard_label=predicted_hard_labels,
+                             elected_hard_label=elected_hard_labels,
+                             predicted_soft_labels=collect(eachrow(predicted_soft_labels)),
+                             votes)
+    observation_table = DataFrame([ObservationRow(r) for r in eachrow(observations)])
```
Comment (ericphanson) [inline src/row.jl:186]: This is probably how I'd write this, to not need the DataFrames dependency:
```suggestion
    votes_itr = has_value(votes) ? eachrow(votes) : (missing for _ in 1:length(predicted_hard_labels))
    predicted_soft_labels_itr = eachrow(predicted_soft_labels_itr)
    if !(length(predicted_hard_labels) == length(predicted_soft_labels_itr) == length(elected_hard_labels) == length(votes_itr))
        throw(DimensionMismatch("Inputs do not all have the same number of observations"))
    end
    observation_table = map(predicted_hard_labels, elected_hard_labels, predicted_soft_labels_itr, votes_itr) do predicted_hard_label, elected_hard_label, predicted_soft_label, votes
        ObservationRow(; predicted_hard_label, elected_hard_label, predicted_soft_label, votes)
    end
```
Now we return a `Vector` of `ObservationRow`s, which is still a valid table. Then if the caller wants a column table, they can just call `Tables.columns` on it.

**Review by ericphanson — APPROVED** [review]
This looks great! Very excited to see the tabular interface. I think though we can avoid the DataFrames dependency, using just Tables.jl (which is already a dependency). I left some comments to that affect. LGTM besides that.

File: src/row.jl (hunk 2 above)
Comment (hannahilea) [reply src/row.jl:186]: oh, awesome. i'm going to leave DataFrames in the tests, I think, but will move it out of the main package!

Comment (ericphanson) [reply src/row.jl:186]: Ok, sounds good!

**Pull request merged**